packages/p2p/src/components/my-ads/my-ads-table.jsx

```diff
@@ -0,0 +1,124 @@
+import React                  from 'react';
+import PropTypes              from 'prop-types';
+import { Table, Button }      from 'deriv-components';
+import { localize }           from 'deriv-translations';
+import { InfiniteLoaderList } from '../table/infinite-loader-list.jsx';
+
+// TODO: replace with API response
+const mock_response = {
+    ad_id          : 'Buy 000023434',
+    amount         : 'BTC 0.00001234',
+    price          : '34,000 MYR/BTC',
+    min_transaction: 'MYR 50',
+    payment_method : 'Bank transfer',
+};
+const initial_data = [
+    { ...mock_response },
+    { ...mock_response },
+    { ...mock_response },
+    { ...mock_response },
+];
+
+const getMockData = () => {
+    return new Promise((resolve) => {
+        setTimeout(function() {
+            resolve([
+                { ...mock_response }, { ...mock_response }, { ...mock_response }, { ...mock_response },
+            ]);
+        }, 300);
+    });
+};
+
+const headers = [
+    { text: localize('Ad ID')  },
+    { text: localize('Amount') },
+    { text: localize('Price') },
+    { text: localize('Min transaction') },
+    { text: localize('Payment Method') },
+];
+
+const RowComponent = React.memo(({ data, style }) => (
+    <div style={style}>
+        <Table.Row>
+            <Table.Cell>{data.ad_id}</Table.Cell>
+            <Table.Cell>{data.amount}</Table.Cell>
+            <Table.Cell>{data.price}</Table.Cell>
+            <Table.Cell>{data.min_transaction}</Table.Cell>
+            <Table.Cell>{data.payment_method}</Table.Cell>
+            <Table.Cell>
+                <div style={{
+                    marginRight: '9px',
+                }}>
+                    <Button secondary small>{localize('Edit')}</Button>
+                </div>
+                <Button secondary small>{localize('Delete')}</Button>
+            </Table.Cell>
+        </Table.Row>
+    </div>
+));
+RowComponent.propTypes = {
+    data : PropTypes.object,
+    style: PropTypes.object,
+};
+RowComponent.displayName = 'RowComponent';
+
+export class MyAdsTable extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            items                 : initial_data,
+            is_loading_more_items : false,
+            has_more_items_to_load: true,
+            width                 : null,
+            height                : null,
+        };
+        this.table_container_ref = React.createRef();
+    }
```

Comment:
Since we are not using props here, can we use: 
```jsx
export class MyAdsTable extends React.Component {
state = {
    items                 : initial_data,
    is_loading_more_items : false,
    has_more_items_to_load: true,
    width                 : null,
    height                : null,
}

table_container_ref = React.createRef();
```

Reply:
true 👍